Source/Core/Core/FreeLookManager.cpp

```diff
@@ -146,8 +147,15 @@ ControllerEmu::ControlGroup* FreeLookController::GetGroup(FreeLookGroup group) c
 
 void FreeLookController::Update()
 {
+  const bool prev_input_gate = ControlReference::GetInputGate();
+
   if (!g_freelook_camera.IsActive())
-    return;
+  {
+    // We must GetState even when inactive in case hotkey @() expressions were pressed during
+    // deactivation, otherwise they will never release their button suppressions.
+    // Turning the input gate off will block all input and release any hotkeys.
+    ControlReference::SetInputGate(false);
```

Comment:
I think we still need a return after this?  The idea is that we don't want to do any current or future logic if we are inactive.

Reply:
Could we do:

```
if (g_freelook_camera.IsActive())
{
  ControlReference::SetInputGate(true);
}
else
{
  ControlReference::SetInputGate(false);
  return;
}
```

That would preserve the early out behavior

Reply:
This won't work. It can't return. The current fix is closing the input gate AND reading the inputs so they release their suppressions.

Reply:
Ahhhhh, I missed that.

Reply:
Not sure if this makes sense:

```
if (g_freelook_camera.IsActive())
{
  ControlReference::SetInputGate(true);
}
else
{
  const bool prev_input_gate = ControlReference::GetInputGate();
  ControlReference::SetInputGate(false);
  if (prev_input_gate)
    DoUpdate();  // Flush suppressions...
  return;
}

DoUpdate();  // This function is essentially 'Update' now without the active check
```

Reply:
That's roughly my logic, but this suggestion has potential to get stuck hotkeys on focus loss situations if you don't call `DoUpdate` when the input gate was closed.

Reply:
I think the above snippet should always close it because we will always transition between an input gate of true to false.  The catch is we only call `DoUpdate()` once to do the flush.  Your current code is calling it every single time.

The problem with this is when there are non-hotkey buttons mapped.  An example today would be assigning Free Look speed increase to a gamepad button.  With your code, if I hit that gamepad button, it'll trigger even if Free Look is inactive.  The current code only triggers if Free Look is active.

Reply:
Maybe I'm misunderstanding the input gate.  Does turning it to false block _all_ input or just hotkeys?